precompose/src/commonTest/kotlin/moe/tlaster/precompose/navigation/BackStackManagerTest.kt

```diff
@@ -260,6 +260,38 @@ class BackStackManagerTest {
         assertEquals("1", backStacks[1].pathMap["id"])
     }
 
+    @Test
+    fun testMultipleLaunchSingleTopWithPopUpTo() {
+        val manager = BackStackManager()
+        manager.init(
+            RouteGraph(
+                "foo/bar",
+                listOf(
+                    TestRoute("foo/bar", "foo/bar"),
+                    TestRoute("foo/baz", "foo/baz"),
+                )
+            ),
+            StateHolder(),
+            TestLifecycleOwner()
+        )
+
+        fun navigate(path: String, navOptions: NavOptions) {
+            val previousEntry = manager.backStacks.value.lastOrNull()
+            manager.push(path, navOptions)
+            // Mark the previous entry as inactive to simulate the lifecycle change by the NavHost
+            previousEntry?.inActive()
```

Comment:
When a destination is destroyed due to the navigation, if its state is not inactive, we mark it to be destroyed on the next lifecycle inactivation, and we disable navigation, to make the test work with this behavior I added this snippet to simulate what a NavHost would do.